ql/src/java/org/apache/hadoop/hive/ql/ddl/view/drop/DropViewAnalyzer.java

```diff
@@ -50,7 +50,7 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     Table view = getTable(viewName, throwException);
     if (view != null) {
       inputs.add(new ReadEntity(view));
-      outputs.add(new WriteEntity(view, WriteEntity.WriteType.DDL_EXCLUSIVE));
+      outputs.add(new WriteEntity(view, WriteEntity.WriteType.DDL_NO_LOCK)); // see AcidUtils.needsLock(output)
```

Comment:
What about materialized view-s where the view could be used for optimising the query and it could be used as a source instead of the original source table?

Or that is a different story?

Reply:
there are 2 separate PR for that, already reviewed by @kasakrisz